src/services/assetService.ts

```diff
@@ -1,10 +1,27 @@
 import MD5 from "md5.js";
 import _ from "lodash";
+import * as shortid from "shortid";
 import Guard from "../common/guard";
-import { IAsset, AssetType, IProject, IAssetMetadata, AssetState } from "../models/applicationState";
+import { IAsset, AssetType, IProject, IAssetMetadata,
+         AssetState, IRegion, RegionType } from "../models/applicationState";
 import { AssetProviderFactory, IAssetProvider } from "../providers/storage/assetProviderFactory";
 import { StorageProviderFactory, IStorageProvider } from "../providers/storage/storageProviderFactory";
 import { constants } from "../common/constants";
+import HtmlFileReader from "../common/htmlFileReader";
+import { TFRecordsReader } from "../providers/export/tensorFlowRecords/tensorFlowReader";
+import { FeatureType } from "../providers/export/tensorFlowRecords/tensorFlowBuilder";
+// tslint:disable-next-line:no-var-requires
+const TagColors = require("../react/components/common/tagsInput/tagColors.json");
+
+interface ITFRecordObjectArray {
```

Comment:
It doesn't feel like this interface belongs in this generic `assetService` file

Reply:
I used tuple first but unfortunately TS/JS doesn't support named tuple properties and using the array syntax was really really bad to see in the code so I decided to use a JS object and as we're on TS I created this interface.

I agree it's not ideal to have this here but indeed this interface it's just needed for a private function in the assetService.

I can move the interface on a different file, as well as the getTFRecordObjectArrays() method that return this interface/object but I don't really want to put these in any TFRecords functions that I already have (like Builder, Reader, Helpers) as this code is specific to the way VoTT use TFRecords and not about how TFRecord internal implementation works.  This is specially true if we want to ship the generic TFRecords stuff on a separated npm package.

Do you think it's better to create this separated file for this interface and method and store it in the same services folder ?  I'm not 100% sure it's better.

Reply:
OK - Let's move it into our core models in `applicationState`.  Let's also rename to make it more clear... `ITFRecordObjectArray` has array in the name but its not an array, but rather has members which some members are an array.  Maybe just `ITFRecord` ?

Reply:
ITFRecordMetadata now, in applicationState